com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfGeometryCollision.cs

```diff
@@ -103,16 +103,42 @@ private static GameObject CreateMeshColliderRuntime(Link.Geometry.Mesh mesh)
         private static GameObject CreateCylinderCollider()
         {
             GameObject gameObject = new GameObject("Cylinder");
-            MeshCollider meshCollider = gameObject.AddComponent<MeshCollider>();
+            MeshFilter meshFilter = gameObject.AddComponent<MeshFilter>();
 
             Link.Geometry.Cylinder cylinder = new Link.Geometry.Cylinder(0.5, 2); //Default unity cylinder sizes
 
-            meshCollider.sharedMesh = CreateCylinderMesh(cylinder);
-            meshCollider.convex = true;
+            meshFilter.sharedMesh = CreateCylinderMesh(cylinder);
+            ConvertCylinderToCollider(meshFilter);
 
             return gameObject;
         }
 
+        private static void ConvertCylinderToCollider(MeshFilter filter)
+        {
+            GameObject go = filter.gameObject;
+            VHACD decomposer = go.AddComponent<VHACD>();
```

Comment:
just curious: why do we always need to use the VHACD decomposer to create the cylinder collider?

Reply:
I agree. since cylinder is always going to be convex, it might be better and also a bit more efficient to use the unity decomposition method for both cases regardless of whether user has selected vhacd or unity.